### ISSUE TITLE
Can't replicate the results

### ISSUE THREAD

**Issue by andreaavo9** [Sirui-Xu/STARS#1]
Hello,
running the code with the default settings i can't replicate the results of the given pretrain.
Have you changed some parameters in order to obtain SotA results with the deterministic model?
The default epochs number is 50, but i suppose you have increased the number of epochs.
Thanks a lot!

**Comment by Sirui-Xu** [comment]
Hi, 

I assume that the results you reproduce are not very different from ours (for no other more essential reason), in which case you can reduce the number of training epochs, for example to 40.

Best regards.

**Comment by andreaavo9** [comment]
Hi,
thanks for the answer.
With your pretrain running in test mode i get 108.4119 mm of error (test on subject 5)
Training the model from scratch, the best result that i had is 113.2632 (subject 5 too).

The performance gap is quite important, obviusly i'm taking the checkpoint with the minimum error in valid test during the train, using the default code. The only thing that i changed is the path of the dataset and the output frames from 10 to 25 to measure the performances with 25 frames of prediction. I even tried to train using subject number 5 as valid test, but the results are not changing, still the same results with a variation of +-1mm of error.

I don't think it's a version problem due to the big gap, even if the code don't present any seed to replicate the results.
By the way, i'm running the code on a RTX 3090 with cuda 11.6 and cudnn 8

It would be very helpful having your configuration setup that you used to reach SotA results, in order to have a better understanding of the code. Can you please check if you are able to get the results written on the paper with this version of the code? maybe it's just a versioning problem.
Thanks a lot for your time!

**Comment by Sirui-Xu** [comment]
Hi,

Could you provide your command for training? 

Following the same settings in the paper, the command to predict local motion with 25 frames should be, 
python main.py --input_n 10 --output_n 25 --skip_rate 2 --n_pre 35 --data_dir ./data --version long

Best.

**Comment by andreaavo9** [comment]
I launched the training with the same exact settings that you wrote (even the same relative path though!).
Can you kindly launch the training with the fresh code of the repo to check if the results are the same?

**Comment by Sirui-Xu** [comment]
Yes, we are running the code based on current version.

Regarding what you mentioned, "obviously i'm taking the checkpoint with the minimum error in valid test during the train", that's not actually what we did. As I recall we trained directly for 40 (maybe 50) epochs and used the checkpoint for the last epoch.

**Comment by andreaavo9** [comment]
Sure, i ran the tests even on the pretrain given every 10 epochs. So the model obtained at epoch 40 and 50, but with these pretrains the gap is even worse (7/8 mm gap between the results of the paper), that's why i didn't mentioned that results.
If you are running the code, in a couple hours we should have the actual results.

**Comment by Sirui-Xu** [comment]
Hi,

Thanks for bringing this to my attention. I have discovered that there is indeed a version issue, which results in much worse performance compared to the results in an old version: https://github.com/Sirui-Xu/STARS/tree/708c1edac44f1f18c346b50e0fb08ff1969d4977

We are working to figure out the cause and will make the necessary updates. In the meantime, you can use this old version if you'd like.

Many thanks.

**Comment by Sirui-Xu** [comment]
Hi,

The issue has been fixed.

Best.

**Comment by andreaavo9** [comment]
Thank you very much!
There is still an error, line 84 of the code, in the dataloader.

**Comment by Sirui-Xu** [comment]
Thanks a lot!

It should be fixed by now.

**Comment by andreaavo9** [comment]
Hello,
i tested the code with the same tests that i did for the previous version of the code and i still cannot replicate the results.
With this new version the error is lower, i got 112.91 mm in testing.
I tried to retrain the model with your settings, i tested every checkpoint (not only the last one) including the best one.
I also retrained using the test set as the validation set (in order to get the best model pretrain that suits the test set, even if it's "cheating" but just to try every configuration that can help the model).
The error given is far from the 108 mm of the paper, don't know what else to try.

I assume that in this new version you tried to train with this exactly version of the code and that you get Sota results.
Can you please share all the details? Thank you very much!

**Comment by Sirui-Xu** [comment]
Hi,

I re-ran the experiment on the existing code.

For training, I used the following command:
`python main.py --input_n 10 --output_n 25 --skip_rate 2 --n_pre 35 --data_dir ../../dataset/ --version long `

The log file can be found here: [log](https://github.com/Sirui-Xu/STARS/blob/9e5af4201db78b0da48b249fd7d8c25925dac4aa/deterministic/checkpoints/CKPT_3D_H36M/h36_3d_10_25_2_35_long_ckpt_local_train.log)

For testing, I tested on the model at epoch 50: 

`python main.py --input_n 10 --output_n 25 --skip_rate 2 --n_pre 35 --data_dir ../../dataset/ --version long --mode test --model_path checkpoints/CKPT_3D_H36M/h36_3d_25frames_ckpt_50 --
test_output_n 25`

The log file can be found here: [log](https://github.com/Sirui-Xu/STARS/blob/9e5af4201db78b0da48b249fd7d8c25925dac4aa/deterministic/checkpoints/CKPT_3D_H36M/h36_3d_10_25_2_35_long_ckpt_local_test.log).  The results are even slightly better than those reported in the paper.

If you have any further questions or require additional information, please feel free to contact me.

Best.

**Comment by Sirui-Xu** [comment]
Also, I re-ran more experiments with different seeds based on the existing code and tested on epoch 50. The error appears between 108.1-108.5

Best.

**Comment by andreaavo9** [comment]
Thank you very much for the log files, it was very useful!
I discovered that with the same code, when i load the training set it loads 180077 samples, but yours is loading 181584 samples.
Probably there is a problem with my dataset folder and not with your architecture.
Hopefully, i will be able to find the problem with my tests with this last step!
I will let you know what's the problem with my setup, maybe it will help other people!
Thanks again for your help!

**Comment by andreaavo9** [comment]
I downloaded again the dataset from the link in the repo (http://www.cs.stanford.edu/people/ashesh/h3.6m.zip) and the latest fresh code. 
As reported in your log file, you are loading more sample than the normal dataset linked.
I'm supposing that the lower error is caused by this problem, can you please download again the dataset and retrain the model?

Let me know the results!
Best.

**Comment by Sirui-Xu** [comment]
Thank you for bringing this to my attention. There seems to be a slight difference in the dataset. I will investigate and get back to you.

Best regards.

**Comment by andreaavo9** [comment]
Thank you very much!

Best.

**Comment by Sirui-Xu** [comment]
Hi,

We found some issues when we extracted the poses from the source, resulting in inconsistencies with the standard h3.6m dataset used in the deterministic predictions. The new log file can be found here: [train log](https://github.com/Sirui-Xu/STARS/blob/f94ce7a47209c95ed061888dbdab29e147e69ecd/deterministic/checkpoints/CKPT_3D_H36M/h36_3d_10_25_2_35_long_ckpt_local_train.log)[test log](https://github.com/Sirui-Xu/STARS/blob/f94ce7a47209c95ed061888dbdab29e147e69ecd/deterministic/checkpoints/CKPT_3D_H36M/h36_3d_10_25_2_35_long_ckpt_local_test.log). We've updated the repo and will update the standard results in the paper.

We have checked that the diverse prediction part is fine and not affected by this issue.

Thanks again for your help!
Best.
